dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/VelocityWebUtil.java

```diff
@@ -0,0 +1,151 @@
+        this.req = context.getRequest();
+        this.ctx = context.getVelocityContext();
+        this.debug = Config.getBooleanProperty(_logVariable, false);
+
+        if (!this.debug) {
+            try {
+                if (ctx.get("request") != null) {
+                    if (((HttpServletRequest) ctx.get("request")).getParameter(_logVariable) != null
+                            || (req.getSession(false) != null && req.getSession(false)
+                                .getAttribute(_logVariable) != null)) {
+                        this.debug = true;
+
+
+                        if ("false".equals(req.getParameter(_logVariable))) {
+                            this.debug = true;
+                            req.getSession()
+                                .removeAttribute(_logVariable);
+                        } else {
+                            req.getSession()
+                                .setAttribute(_logVariable, "true");
+                        }
+
+
+
+                    }
+                }
+            } catch (Exception e) {
+                Logger.debug(VelocityWebUtil.class, e.getMessage(), e);
+            }
+        }
+
+    }
+
+    public String mergeTemplate(String templatePath) throws ResourceNotFoundException, ParseErrorException, Exception {
+
+        VelocityEngine ve = VelocityUtil.getEngine();
+        Template template = null;
+        StringWriter sw = new StringWriter();
+        String threadName = Thread.currentThread()
+            .getName();
+        if (this.debug) {
+            Logger.info(VelocityWebUtil.class, _logVariable + ": " + templatePath);
+        }
+        try {
+            String newThreadName = (threadName.contains("{")) ? threadName.replaceAll("\\{[^\\}]*\\}", "{" + templatePath + "}")
+                    : threadName + " {" + templatePath + "}";
+            Thread.currentThread()
+                .setName(newThreadName);
+            template = ve.getTemplate(templatePath);
+
+            template.merge(ctx, sw);
+            return sw.toString();
+        } catch (ParseErrorException e) {
+            if (null != ctx) {
+
+                /*
+                 * In Edit mode we are allow to fail and be noisy, but on Preview and Live mode we
+                 * just want to continue with the render of the page.
+                 */
+                PageMode mode = (PageMode) ctx.get(WebKeys.PAGE_MODE_SESSION);
+
+                if (null == mode || mode != PageMode.EDIT_MODE) {
+                    Logger.error(this.getClass(), "Error parsing elements", e);
+                    return sw.toString();
+                }
+            }
+
+            throw e;
+        } finally {
+            Thread.currentThread()
+                .setName(threadName);
+        }
+
+
+    }
+
+
+    public void mergeTemplate(String templatePath, HttpServletResponse response)
+            throws ResourceNotFoundException, ParseErrorException, Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.apache.velocity.exception.ResourceNotFoundException' which is a subclass of 'java.lang.Exception'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.apache.velocity.exception.ParseErrorException' which is a subclass of 'java.lang.Exception'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)